### DISCUSSION TITLE
[quantization] is there any plan to support 6bits quantization , as 6 bits quantization more efficient than 8 bits on arm cpu

### DISCUSSION THREAD

**Discussion by tianylijun** [apache/incubator-mxnet#15432] [Q&A]
Is there any plan to support 6bits quantization , as data overflow risk, 6 bits quantization more efficient than 8 bits on ARM CPU.

[a comment by an automated account is not shown]

**Comment by pengzhao-intel** [comment]
Thanks for the proposal. 
Do you have any technical details about 6bits is more efficient than 8bits?

**Comment by tianylijun** [comment]
> Thanks for the proposal.
> Do you have any technical details about 6bits is more efficient than 8bits?

As current ARM SIMD do not support int32 += int8xint8（except Cotex A55，A75)，unfortunate 8bits MAC will overflow during 4X4 sgemm block(127x127x4 exceed int16 range )，so int8 need first convert to int16 then do int32 += int16xint16. but 6 bits will not overflow in 4x4 sgemm block，so 6 bits can use directly use int16+= int8xint8 MAC，more efficient than 8 bits

**Comment by pengzhao-intel** [comment]
@tianylijun thanks for the explanation. The tensorcore compute the 4x4 GEMM with INT8 and seems it can handle the overflow. Did you have a chance to look into the difference?
